lib/shopify_app/controller_concerns/login_protection.rb

```diff
@@ -8,7 +8,9 @@ module LoginProtection
     include ShopifyApp::SanitizedParams
 
     included do
-      if ancestors.include?(ShopifyApp::RequireKnownShop || ShopifyApp::EnsureInstalled)
+      if defined?(ShopifyApp::RequireKnownShop) &&
+          defined?(ShopifyApp::EnsureInstalled) &&
+          ancestors.include?(ShopifyApp::RequireKnownShop || ShopifyApp::EnsureInstalled)
```

Comment:
We wanted to put this whole chunk of code in a private method but apparently the whole module doesn't get loaded when using `included`? 
```shell
klenotiw@Bills-MBP shopify_app % bundle exec rake test
rake aborted!
NameError: undefined local variable or method `incompatible_concerns' for AuthenticatedTest::AuthenticatedTestController:Class
/Users/klenotiw/src/github.com/Shopify/shopify_app/lib/shopify_app/controller_concerns/login_protection.rb:11:in `block in <module:LoginProtection>'
/Users/klenotiw/.gem/ruby/3.0.3/gems/activesupport-6.1.7/lib/active_support/concern.rb:136:in `class_eval'
/Users/klenotiw/.gem/ruby/3.0.3/gems/activesupport-6.1.7/lib/active_support/concern.rb:136:in `append_features'
/Users/klenotiw/src/github.com/Shopify/shopify_app/app/controllers/concerns/shopify_app/ensure_has_session.rb:9:in `include'
/Users/klenotiw/src/github.com/Shopify/shopify_app/app/controllers/concerns/shopify_app/ensure_has_session.rb:9:in `block in <module:EnsureHasSession>'
/Users/klenotiw/.gem/ruby/3.0.3/gems/activesupport-6.1.7/lib/active_support/concern.rb:136:in `class_eval'
/Users/klenotiw/.gem/ruby/3.0.3/gems/activesupport-6.1.7/lib/active_support/concern.rb:136:in `append_features'
/Users/klenotiw/.gem/ruby/3.0.3/gems/activesupport-6.1.7/lib/active_support/concern.rb:133:in `include'
/Users/klenotiw/.gem/ruby/3.0.3/gems/activesupport-6.1.7/lib/active_support/concern.rb:133:in `block in append_features'
/Users/klenotiw/.gem/ruby/3.0.3/gems/activesupport-6.1.7/lib/active_support/concern.rb:133:in `each'
/Users/klenotiw/.gem/ruby/3.0.3/gems/activesupport-6.1.7/lib/active_support/concern.rb:133:in `append_features'
/Users/klenotiw/src/github.com/Shopify/shopify_app/test/controllers/concerns/authenticated_test.rb:7:in `include'
/Users/klenotiw/src/github.com/Shopify/shopify_app/test/controllers/concerns/authenticated_test.rb:7:in `<class:AuthenticatedTestController>'
/Users/klenotiw/src/github.com/Shopify/shopify_app/test/controllers/concerns/authenticated_test.rb:6:in `<class:AuthenticatedTest>'
/Users/klenotiw/src/github.com/Shopify/shopify_app/test/controllers/concerns/authenticated_test.rb:5:in `<top (required)>'
/Users/klenotiw/.gem/ruby/3.0.3/gems/zeitwerk-2.6.4/lib/zeitwerk/kernel.rb:38:in `require'
/Users/klenotiw/.gem/ruby/3.0.3/gems/zeitwerk-2.6.4/lib/zeitwerk/kernel.rb:38:in `require'
/Users/klenotiw/.gem/ruby/3.0.3/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:332:in `block in require'
/Users/klenotiw/.gem/ruby/3.0.3/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:299:in `load_dependency'
/Users/klenotiw/.gem/ruby/3.0.3/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:332:in `require'
/Users/klenotiw/.gem/ruby/3.0.3/gems/railties-6.1.7/lib/rails/test_unit/runner.rb:50:in `block in load_tests'
/Users/klenotiw/.gem/ruby/3.0.3/gems/railties-6.1.7/lib/rails/test_unit/runner.rb:50:in `each'
/Users/klenotiw/.gem/ruby/3.0.3/gems/railties-6.1.7/lib/rails/test_unit/runner.rb:50:in `load_tests'
/Users/klenotiw/.gem/ruby/3.0.3/gems/railties-6.1.7/lib/rails/test_unit/runner.rb:39:in `run'
/Users/klenotiw/.gem/ruby/3.0.3/gems/railties-6.1.7/lib/rails/test_unit/runner.rb:35:in `rake_run'
/Users/klenotiw/.gem/ruby/3.0.3/gems/railties-6.1.7/lib/rails/test_unit/testing.rake:16:in `block in <top (required)>'
/Users/klenotiw/.gem/ruby/3.0.3/gems/rake-13.0.6/exe/rake:27:in `<top (required)>'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/lib/bundler/cli/exec.rb:58:in `load'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/lib/bundler/cli/exec.rb:58:in `kernel_load'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/lib/bundler/cli/exec.rb:23:in `run'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/lib/bundler/cli.rb:484:in `exec'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/lib/bundler/cli.rb:31:in `dispatch'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/lib/bundler/cli.rb:25:in `start'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/exe/bundle:48:in `block in <top (required)>'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/lib/bundler/friendly_errors.rb:103:in `with_friendly_errors'
/Users/klenotiw/.gem/ruby/3.0.3/gems/bundler-2.3.4/exe/bundle:36:in `<top (required)>'
/Users/klenotiw/.gem/ruby/3.0.3/bin/bundle:23:in `load'
/Users/klenotiw/.gem/ruby/3.0.3/bin/bundle:23:in `<main>'
Tasks: TOP => test
(See full trace by running task with --trace)
klenotiw@Bills-MBP shopify_app % 
```
```ruby
    def incompatible_concerns
      if defined?(ShopifyApp::RequireKnownShop) &&
          defined?(ShopifyApp::EnsureInstalled) &&
          ancestors.include?(ShopifyApp::RequireKnownShop || ShopifyApp::EnsureInstalled)
        message = <<~EOS
          We detected the use of incompatible concerns (RequireKnownShop/EnsureInstalled and LoginProtection) in #{name},
          which may lead to unpredictable behavior. In a future release of this library this will raise an error.
        EOS
        ShopifyApp::Logger.deprecated(message, "22.0.0")
      end
    end

```
This method was put at the bottom of the file.